### DISCUSSION TITLE
Classic Explorer L10N cache?

### DISCUSSION THREAD

**Discussion by aubymori** [Open-Shell/Open-Shell-Menu#1728] [Q&A] [answered]
In my ExplorerL10N.ini, I've edited the status bar strings to display `%s object(s)` instead of `%s items` and so on. For whatever reason, I just CANNOT get this to apply, at all. I've tried:
- Clearing cached information through start menu settings
- Restarting Explorer
- Restarting my whole computer
- Recompiling ClassicExplorer32.dll and ClassicExplorer64.dll to change the fallback strings
- Renaming the ini to get it to use these fallback strings

Yet no matter what I do, **THIS** is what I see when I open Explorer.
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/9ed8f53f-ac34-4ced-81ad-a54ee66589c0)
I've deduced that it *must* be some ungodly cache or something it's getting this god forsaken string from, but I can't be bothered to read hard enough into the source code to figure it out. Could anyone tell me where this cache is so I can purge it?

**Comment by ge0rdi** [comment]
> Yet no matter what I do, THIS is what I see when I open Explorer.

Couldn't it be that you are looking at standard Windows status bar?
Do you have `Open-Shell`'s status bar enabled in `Classic Explorer Settings` -> `Status bar`?

What is your Windows version?

> I've deduced that it must be some ungodly cache ...

There is nothing like that.
Simple edit of `ExplorerL10N.ini` (section with language you are using) and then restart of Explorer should change those strings (works for me as expected).

**Comment by aubymori** [comment]
> Couldn't it be that you are looking at standard Windows status bar?

Nope, I'm 100% sure this is the Open Shell status bar.

> There is nothing like that.
> Simple edit of `ExplorerL10N.ini` (section with language you are using) and then restart of Explorer should change those strings (works for me as expected).

Yeah well, for whatever reason it just did absolutely nothing for me. I made sure I edited the language that was in use too (en-US).

**Reply by ge0rdi** [reply]
> Nope, I'm 100% sure this is the Open Shell status bar.

Could you add screenshot with whole status bar from Explorer window with some local folder opened?
There should be something like `(Disk free space ...)` displayed.

> Yeah well, for whatever reason it just did absolutely nothing for me. I made sure I edited the language that was in use too (en-US).

So you edited `C:\Program Files\Open-Shell\ExplorerL10N.ini`, right?
In `[en-US]` section you edited `Status.Item` and `Status.Items`, saved and then restarted Explorer, right?

What is your Windows version again?

**Comment by aubymori** [comment]
> Could you add screenshot with whole status bar from Explorer window with some local folder opened?

Not at my computer right now but I can confirm that it does display that and is the Open Shell status bar (default status bar does not look right in classic theme).

> So you edited C:\Program Files\Open-Shell\ExplorerL10N.ini, right?
In [en-US] section you edited Status.Item and Status.Items, saved and then restarted Explorer, right?

Yeah.

> What is your Windows version again?

Windows 10 21H2 (10.0.19044.3448)

**Reply by ge0rdi** [reply]
> Not at my computer right now but I can confirm that it does display that and is the Open Shell status bar (default status bar does not look right in classic theme).

Please, try to add that screenshot when you can.
Also check language in `Classic Explorer Settings` -> `Language`. To make sure `Open-Shell` uses `en-US`.

What `Open-Shell` version do you have? Latest one? (4.4.191)

**Reply by aubymori** [reply]
I am using Open Shell 4.4.190 since I have a patched version of StartMenuDLL for that version specifically.

**Reply by aubymori** [reply]
@ge0rdi 
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/65454c03-5c36-4fd3-b71d-bec04edc812f)
(It says 4.4.1000 since I compiled from source, but my codebase is based on that of 4.4.190. Also, I did not modify any code relating to localization, other than changing the fallback string for status bar to my desired string).

**Reply by ge0rdi** [reply]
> It says 4.4.1000 since I compiled from source

Will be better to use official one. You wrote that changing those callbacks and recompiling didn't help.

What about the screenshot of status bar from some local folder opened?

Also you seem to use some _classic theme_.
Is this the default one in Windows or do you use another customization software?

**Reply by aubymori** [reply]
> What about the screenshot of status bar from some local folder opened?

Old, but here.
![image-6-1](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/e3c169df-6cdd-4136-a03a-e0832710962b)

> Also you seem to use some classic theme.
Is this the default one in Windows or do you use another customization software?

Default one in Windows, enabled through ClassicThemeTray.

**Reply by ge0rdi** [reply]
> Old, but here.

Not sure what you mean by "old". But can you provide screenshot with actual `Open-Shell` you are running?
That one looks like `Open-Shell` status bar.
So I have no clue why changing of `ExplorerL10N.ini` shouldn't work :(

**Reply by aubymori** [reply]
> Not sure what you mean by "old".

I mean this screenshot is from like a week ago, since I'm not at my computer (yet again) and this is the most recent one I could find.

> But can you provide screenshot with actual `Open-Shell` you are running?

What exactly do you mean?

**Reply by ge0rdi** [reply]
> What exactly do you mean?

I just wanted to see whether the status bar looks like that now, not in the past.

Anyways, it seem there is something odd on your machine. I have really no clue what :(

As I wrote, there is no caching of status bar text so those localization strings should be read when Explorer starts.
Also if you changed those default translations [here](https://github.com/Open-Shell/Open-Shell-Menu/blob/master/Src/ClassicExplorer/ExplorerBHO.cpp#L371), recompiled and replaced `ClassicExplorer64.dll` in install folder and it didn't help then there is something really weird going on.

**Reply by aubymori** [reply]
@ge0rdi 
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/bb5bbcd7-a049-494a-b9ed-bbddaa986cfd)
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/1c310e9b-7095-4379-b1c1-41ea1686fcfb)
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/aed18e5b-f318-4197-989d-aa599ec19ae3)
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/8e5425fb-6a9c-4ff8-a77b-731800b61562)
yeah, i have no idea what could possibly be going on here.

**Reply by ge0rdi** [reply]
> yeah, i have no idea what could possibly be going on here.

Could you try to change some `Open-Shell` Classic Explorer settings for status bar? Like disable `Show free space and file size` option and then open new Explorer window as before (`C:\` drive) to see if the `Disk free space` information will be there or not.
This way you will confirm that it is really `Open-Shell`'s status bar.

**Reply by aubymori** [reply]
I did that earlier before taking the screenshot and I can confirm, yes, it is Open-Shell's status bar.

**Reply by aubymori** [reply, marked as answer]
@ge0rdi ok i finally fixed it
for some fucking reason it was using classicexplorer64.dll from the build directory instead of from open-shell program files
![image](https://github.com/Open-Shell/Open-Shell-Menu/assets/44238627/dc7e8c65-0ca2-40af-9680-bccf8b147edf)
